### PR TITLE
Add hooks support for WebSub (formerly PubSubHubbub) protocol

### LINKED ISSUE #1265: Support PubSubHubbub API for creating hooks
Are there any plans to support [the PubSubHubbub API for creating webhooks](https://developer.github.com/v3/repos/hooks/#pubsubhubbub)?

I could try to create a PR for implementing this (I'm hoping to use it for implementing https://github.com/terraform-providers/terraform-provider-github/issues/275) if there is any chance of it being accepted.

### PR DESCRIPTION
**Pull request by YuyaAbo**
resolves #1265

_Note: PubSubHubbub seems to have been renamed WebSub when it was adopted by the W3C. So we use the name WebSub in our code.
ref https://github.com/pubsubhubbub/PubSubHubbub_

Added support for API to subscribe and unsubscribe to repository events via WebSub protocol.

GitHub API Docs: https://docs.github.com/en/rest/webhooks#pubsubhubbub
WebSub Docs: https://www.w3.org/TR/websub/

WebSub requires Content-Type to be application/x-www-form-urlencoded, so a new factory function has been added.

> This request MUST have a Content-Type header of application/x-www-form-urlencoded (described in Section 4.10.22.6 [[HTML5](https://www.w3.org/TR/websub/#bib-HTML5)])
> https://www.w3.org/TR/websub/#subscriber-sends-subscription-request

### CODE REVIEW

File: github/repos_hooks.go
Comment (gmlewis) [inline github/repos_hooks.go:216]:
```suggestion
// Unsubscribe lets servers unregister to no longer receive updates when a topic is updated.
```

File: github/repos_hooks.go
Comment (gmlewis) [inline github/repos_hooks.go:228]:
```suggestion
// createWebSubRequest returns a subscribe/unsubscribe request that implements
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @YuyaAbo !
Just a couple tiny tweaks, please, then we should be ready for a second approval before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @YuyaAbo !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

File: github/github.go
Comment (raynigon) [inline github/github.go:444]: It would be nice if this case got tested.

**Review by raynigon — APPROVED** [review]
Can be merged, but it would be nice if the error case test gets added first.

File: github/github.go
Comment (YuyaAbo) [reply github/github.go:444]: There seems to be no case where this `http.NewRequest` returns an error.

There are two cases in which `http.NewRequest` returns an error: when the method is invalid or when the url fails to parse. `NewFormRequest` is set to POST, so the former error does not occur. Also, since url parsing is done in `c.BaseURL.Parse` before this process, if url parsing fails, an error is returned before this process.

Even the already existing NewUploadRequest does not seem to cover this case. Is there a policy for such cases?

Comment (gmlewis) [reply github/github.go:444]: Could you pass in an `iotest.ErrReader` as the body to `NewRequest` to make it fail?
https://pkg.go.dev/testing/iotest#ErrReader

Comment (YuyaAbo) [reply github/github.go:444]: thanks. It didn't work. Just generating a `http.Request` will not cause it to be read by the `iotest.ErrReader`.

Comment (gmlewis) [reply github/github.go:444]: OK, thanks for trying.  I'm OK with that.
Merging.

**Pull request merged**
